app/Platform/Actions/UpdatePlayerPlayedGames.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Actions;
+
+use App\Models\PlayerGame;
+use App\Models\User;
+use App\Platform\Jobs\UpdateGameMetricsJob;
+use App\Platform\Services\GameTopAchieversService;
+use Illuminate\Bus\Batch;
+use Illuminate\Bus\BatchRepository;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Bus;
+
+class UpdatePlayerPlayedGames
```

Comment:
This could use a better name. It's purpose is to update data depending on the player's visibility for all games they've ever played.

Reply:
My first idea was `UpdateGamesAffectedByPlayerVisibilityChange`, but I'm afraid this could be too prescriptive.

`UpdateGameMetricsTouchedByUser`?

Reply:
I went with `UpdateGameMetricsForGamesPlayedByUser`